lib/rails_multitenant/global_context_registry.rb

```diff
@@ -26,21 +26,80 @@ module Current
 
       module ClassMethods
         def current
-          GlobalContextRegistry.fetch(current_registry_obj) { new }
+          GlobalContextRegistry.fetch(current_registry_obj) { __current_default }
+        end
+
+        def current=(object)
+          raise "#{object} is not a #{self}" if object.present? && !object.is_a?(self)
+          GlobalContextRegistry.set(current_registry_obj, object)
+          __clear_dependents!
+        end
+
+        def current?
+          GlobalContextRegistry.get(current_registry_obj).present?
+        end
+
+        def current!
+          current || raise("No current #{name} set")
         end
 
         def clear_current!
           GlobalContextRegistry.delete(current_registry_obj)
         end
 
+        def as_current(object)
+          old_object = current if current?
+          self.current = object
+          yield
+        ensure
+          self.current = old_object
+        end
+
         include RegistryDependentOn
 
+        def provide_default(provider = nil, &block)
+          @default_provider = provider ? provider.to_proc : block
```

Comment:
A class level instance variable won't be inherited e.g. 

```
class Parent
  include Current
  provide_default :new
end

class Child < Parent
end

# Child.current returns nil
```
Is that intentional? Using a [class_attribute](http://guides.rubyonrails.org/active_support_core_extensions.html#class-attributes) from ActiveSupport might be appropriate here.